### DISCUSSION TITLE
Create Account from keys generated from "cardano-cli address key-gen"

### DISCUSSION THREAD

**Discussion by djcyr** [bloxbean/cardano-client-lib#160] [General]
Using the command 		

cardano-cli address key-gen --verification-key-file payment.vkey --signing-key-file payment.skey
		
Is it possible to construct an Account object from the 68 byte cborhex value in the .skey file?

Thanks

**Comment by satran004** [comment]
There is a constructor in account level which takes an account level private key which is usually 96 bytes (includes prv key, chain code, pub key). You can use it to create an account object.

https://github.com/bloxbean/cardano-client-lib/blob/427f3ccbdf20ddbfc063e081f690ad52b2cd4e5c/src/main/java/com/bloxbean/cardano/client/account/Account.java#L149

Here's a testcase for the same 
```
https://github.com/bloxbean/cardano-client-lib/blob/427f3ccbdf20ddbfc063e081f690ad52b2cd4e5c/src/test/java/com/bloxbean/cardano/client/account/AccountTest.java#L79
```

**Address Gen and Signing using verification key and secret key** 

I don't think you can use a payment.vkey or payment.skey directly in Account class. But you can easily generate a payment address from a verification key and similarly sign a transaction using skey.

To generate the payment address from vkey, create a VerificationKey instance with cborhex or just deserialize the json in payment.vkey file.

```
VerificationKey vk = new VerificationKey("58209b0ee7e26318d3675742c8b841b981ae57c37bcbb2cc5625f80606b3256d0814");

HdPublicKey hdPublicKey = new HdPublicKey();
hdPublicKey.setKeyData(vk.getBytes());
Address address = AddressService.getInstance().getEntAddress(hdPublicKey, Networks.testnet());   //Ent address as there's no staking part
``` 

Similarly, to sign a transaction with payment.skey, there is a signer helper method.

```
SecretKey sk = new SecretKey("58208b7dae4189261f1c8b4fcf...");
TxSigner signer = SignerProviders.signerFrom(sk);
Transaction signedTransaction = TxBuilderContext.init(utxoSupplier, protocolParamsSupplier)
                .buildAndSign(txBuilder, signer);
```

The above code snippet is using composable functions to build txn. 

But if you are using low level transaction api / high level txn, you can directly use TransactionSigner.sign() to sign the transasction with a SecretKey. But the recommended way is to use composable functions. :)
